src/schema/sources.ts

```diff
@@ -1121,6 +1138,30 @@ export const resolvers: IResolvers<any, Context> = {
       }
       return res[0];
     },
+    relatedTags: async (_, { sourceId }, ctx): Promise<GQLTagResults> => {
+      const timeThreshold = subDays(new Date(), 90);
+      const keywords = await ctx.con
+        .createQueryBuilder()
+        .from(Post, 'p')
+        .select('pk.keyword, count(pk.keyword) as count')
+        .innerJoin(
+          PostKeyword,
+          'pk',
+          'pk.postId = p.id and pk.status = :status',
+          { status: 'allow' },
+        )
+        .where({ sourceId })
```

Comment:
Do we need to check if source access is allowed?

Reply:
Nah think its fine, doesnt really expose anything and would cost another query 🥲

Reply:
Ok, let's leave it as is for now.